### DISCUSSION TITLE
PostgreSQL multi statement error

### DISCUSSION THREAD

**Discussion by Keda87** [Budibase/budibase#6182] [Help] [answered]
Hi,
I'm having an issue when I need to do a multi-statement query. in my case, I have to do some operations on atomic for data consistency.
```sql
BEGIN;

UPDATE xxx ...;

INSERT INTO yyy ...;

COMMIT;
```

But we got this error
```json
{
    "message":"error: cannot insert multiple commands into a prepared statement",
    "status":400,
    "error":{
        "type":"Error"
    }
}
```

is there any workaround to do this? because I didn't find any clue in the documentation.

Thank you in advanced

**Comment by shogunpurple** [comment, marked as answer]
Hi @Keda87

I'm assuming you are using postgres due to the error message. The error message is exactly as it states - you cannot use multiple semi-colon separated SQL commands in a prepared statement.

More information here:
https://github.com/brianc/node-postgres/issues/83

2 options here:
- Execute the queries one by one, ie. `BEGIN`, query, then `COMMIT` by setting up separate queries through button actions in your app.
- Edit your SQL to be one query. This could be difficult, but you should get a pretty good idea of how to do that here: https://dba.stackexchange.com/questions/218234/how-to-perform-multiple-inserts-into-postgresql-database-in-one-query

Thanks!

**Reply by Keda87** [reply]
Hi @shogunpurple
Thank you for the answer. I Just want to ensure if I use the first option, are there any differences if I define separate query `BEGIN/COMMIT` using query config `Update` or `Write`?
